<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MessageFilterDialog</class>
 <widget class="QDialog" name="MessageFilterDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>495</width>
    <height>348</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Filter/Highlight Messages</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="filterTab">
      <attribute name="title">
       <string>Filter</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="SearchInputPane" name="findPane">
            <property name="whatsThis">
             <string>Type a full or partial message name to narrow down the list</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListView" name="filterListView">
            <property name="whatsThis">
             <string>List of all applicable message types. Toggle checkbox to include or exclude.</string>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::ExtendedSelection</enum>
            </property>
            <property name="verticalScrollMode">
             <enum>QAbstractItemView::ScrollPerPixel</enum>
            </property>
            <property name="horizontalScrollMode">
             <enum>QAbstractItemView::ScrollPerPixel</enum>
            </property>
            <property name="uniformItemSizes">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>6</number>
          </property>
          <item>
           <widget class="QCheckBox" name="otherMessagesCheckBox">
            <property name="whatsThis">
             <string>Include/Exclude unknown messages (ones not in the list).</string>
            </property>
            <property name="text">
             <string>All Others</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="includeAllButton">
            <property name="whatsThis">
             <string>Click to include all message types (ticks all boxes in the list).</string>
            </property>
            <property name="text">
             <string>I&amp;nclude All</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="excludeAllButton">
            <property name="whatsThis">
             <string>Click to exclude all message types (unticks all boxes in the list).</string>
            </property>
            <property name="text">
             <string>E&amp;xclude All</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="includeSelectedButton">
            <property name="whatsThis">
             <string>Click to include all message types that are selected in the list.</string>
            </property>
            <property name="text">
             <string>&amp;Include Selected</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="excludeSelectedButton">
            <property name="whatsThis">
             <string>Click to exclude all message types that are selected in the list.</string>
            </property>
            <property name="text">
             <string>&amp;Exclude Selected</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="highlightTab">
      <attribute name="title">
       <string>Highlight</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTableWidget" name="highlightsTable">
         <property name="whatsThis">
          <string>List of message types to highlight. Double click each colour to change it.</string>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="verticalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <property name="horizontalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>Message</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Text Colour</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Back Colour</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_2" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="addHighlightButton">
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="removeHighlightButton">
            <property name="text">
             <string>Remove</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="dialogButtons">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SearchInputPane</class>
   <extends>QLineEdit</extends>
   <header>ui/custom_widgets/SearchInputPane.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>findPane</tabstop>
  <tabstop>filterListView</tabstop>
  <tabstop>otherMessagesCheckBox</tabstop>
  <tabstop>includeAllButton</tabstop>
  <tabstop>excludeAllButton</tabstop>
  <tabstop>includeSelectedButton</tabstop>
  <tabstop>excludeSelectedButton</tabstop>
  <tabstop>dialogButtons</tabstop>
  <tabstop>highlightsTable</tabstop>
  <tabstop>addHighlightButton</tabstop>
  <tabstop>removeHighlightButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>dialogButtons</sender>
   <signal>accepted()</signal>
   <receiver>MessageFilterDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dialogButtons</sender>
   <signal>rejected()</signal>
   <receiver>MessageFilterDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
